plugins/star_wars_unlimited/swudb.py

```diff
@@ -0,0 +1,117 @@
+from os import path
+from requests import Response, get
+from time import sleep
+from re import sub
+from PIL import Image
+
+SWUDB_FRONT_ART_URL_TEMPLATE = 'https://swudb.com/images/cards/{card_number}.png'
+SWUDB_BACK_ART_URL_TEMPLATE_1 = 'https://swudb.com/images/cards/{card_number}-back.png'
+SWUDB_BACK_ART_URL_TEMPLATE_2 = 'https://swudb.com/images/cards/{card_number}-portrait.png'
+SWUDB_NAME_URL_TEMPLATE = 'https://swudb.com/api/search/{name}{title}?grouping=cards&sortorder=setno&sortdir=asc'
+SWUDB_ART_URL_TEMPLATE = 'https://swudb.com/images/cards/{card_art_ref}'
+
+OUTPUT_CARD_ART_FILE_TEMPLATE = '{deck_index}{card_name}{quantity_counter}.png'
+
+def ping_swudb(query: str) -> bool:
+    r = get(query, headers = {'user-agent': 'silhouette-card-maker/0.1', 'accept': '*/*'})
+
+    sleep(0.15)
+
+    return False if r.status_code == 404 else True
+
+def request_swudb(query: str) -> Response:
+    r = get(query, headers = {'user-agent': 'silhouette-card-maker/0.1', 'accept': '*/*'})
+
+    r.raise_for_status()
+    sleep(0.15)
+
+    return r
+
+def fetch_card(
+    index: int,
+    quantity: int,
+    name: str,
+    title: str,
+    card_number: str,
+    front_img_dir: str,
+    back_img_dir: str,
+):
+    # Query for card art
+    if card_number != '': # Fetch card art by card number
+        card_number_url_parse = sub('_', '/', card_number)
+        front_art = request_swudb(SWUDB_FRONT_ART_URL_TEMPLATE.format(card_number=card_number_url_parse)).content
+        back_art = None
+
+        # Ping to verify the server has the back art in one of the possible naming formats, since we do not know if it does
+        if ping_swudb(SWUDB_BACK_ART_URL_TEMPLATE_1.format(card_number=card_number_url_parse)) is True:
+            back_art = request_swudb(SWUDB_BACK_ART_URL_TEMPLATE_1.format(card_number=card_number_url_parse)).content
+        elif ping_swudb(SWUDB_BACK_ART_URL_TEMPLATE_2.format(card_number=card_number_url_parse)) is True:
+            back_art = request_swudb(SWUDB_BACK_ART_URL_TEMPLATE_2.format(card_number=card_number_url_parse)).content
+    else: # Fetch card art by querying name and title
+        title_query = '' if title == '' else f' title:"{title}"'
+        json = request_swudb(SWUDB_NAME_URL_TEMPLATE.format(name=name, title=title_query)).json()
+        art_url_suffix = sub('.+cards/', '', json.get('variants')[0].get('frontImagePath'))
+        front_art = request_swudb(SWUDB_ART_URL_TEMPLATE.format(card_art_ref=art_url_suffix)).content
```

Comment:
I think it's strange to extract the front image path. At this point, we should be able to get the card number and expansion abbreviation. 

This is related to https://github.com/Alan-Cha/silhouette-card-maker/pull/23/files#r2280967063 because if we did that, then we can reuse the URL here.

```python
json.get('variants')[0].get('expansionAbbreviation')
json.get('variants')[0].get('cardNumber')
```

Is it also necessary to use the `get()` function here?